### PR TITLE
feat: ordered concerning/reassuring feats by contribution size

### PR DESCRIPTION
**Pull request by slebastard**
Marginal additions/modifications to explainability module:
- sorting `expl_selection` lists `select_concerning` and `select_reassuring` so that the lists are ordered by decreasing contribution size
- reversing the direction of radar variables
- grouping categoricals and continuous features of group `retards_paiement` into a single micro_variable `retards_paiement`, or equivalently, using the macro variable `retards_paiement` as the only micro variables for `retards_paiement`
- adding unit tests for functions generating `expl_selection` variable

### CODE REVIEW

**Comment by slebastard** [comment]
Please merge #70 before this PR.
#70 affects the codebase in a broader way, and rebasing `feat/explain_v2` on `develop` after merging `feat/e2e_list_production` will be easier.

**Comment by vviers** [comment]
@slebastard can you please rebase this branch and resolve conflicts before I can review and merge it ? :)

**Comment by slebastard** [comment]
> @slebastard can you please rebase this branch and resolve conflicts before I can review and merge it ? :)

Rebase done, you should be good to go!

File: predictsignauxfaibles/explain.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import pandas as pd
+
+from predictsignauxfaibles.data import SFDataset
+from predictsignauxfaibles.utils import (
+    sigmoid,
+    make_multi_columns,
+)
+
+
+def list_concerning_contributions(entry: pd.Series, thr=0.07):
```
Comment (vviers) [inline predictsignauxfaibles/explain.py]:
```suggestion
def list_concerning_contributions(entry: pd.Series, thr: float =0.07):
```

File: predictsignauxfaibles/explain.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    """
+    From a record containing dot-products of each feature with the weight vector,
+    computes a list of features that contribute to a high risk of failure
+    """
+    masked = entry.where(entry >= 4 * thr)
```
Comment (vviers) [inline predictsignauxfaibles/explain.py:18]: Sorry if I missed this info but why are we multiplying the threshold by +/- 4 ?

File: predictsignauxfaibles/explain.py (hunk 1 above)
Comment (vviers) [inline predictsignauxfaibles/explain.py]: Why is 0.07 the default ? Do we _need_ a default value for `thr` here ?

**Review by vviers — DISMISSED** [review]
Thanks for this oh-so important PR 💯 !
A few non-blocking comments and suggestion, feel free to merge 

Next big step in to document explainability :) i'll create a Trello card

File: predictsignauxfaibles/explain.py (hunk 2 above)
Comment (slebastard) [reply predictsignauxfaibles/explain.py:18]: Yes, there is good theoretical grounding for this, and I will need to document that in detail.
Essentially, the factor 4 stems from the maximum derivative of a sigmoid, which we integrate to select the maximum possible contribution of a group of feature to our score. See picture below, to be documented soon:
sf_logreg_explain_p2

File: predictsignauxfaibles/explain.py (hunk 1 above)
Comment (slebastard) [reply predictsignauxfaibles/explain.py]: Default threshold 0.07 was selected through a first manual fit on examples where the contribution of groups of variables seemed obvious. The threshold will probably be adjusted though, first through testing on small datasets. This evaluation phase will reveal whether or not we need to add a CLI option to specify a threshold for explain features. To follow based on feedback from the front-end team!

**Pull request merged**